### PR TITLE
feat(jsonwebtoken): improve jsonwebtoken types

### PR DESCRIPTION
**Pull request by facundo-viale-olx**

### CODE REVIEW

File: definitions/npm/jsonwebtoken_v8.2.x/flow_v0.56.x-/jsonwebtoken_v8.2.x.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
+
```
Comment (GAntoine) [inline definitions/npm/jsonwebtoken_v8.2.x/flow_v0.56.x-/jsonwebtoken_v8.2.x.js]: Remove this whitespace

**Comment by facundo-viale-olx** [comment]
I was not sure about putting all the new changes in current version ( 8.1.0) or just put them in the a new folder for the last version ( 8.2.0 ). I can merge the version into 8.x.x if you want or any other alternative =)

File: definitions/npm/jsonwebtoken_v8.2.x/flow_v0.56.x-/jsonwebtoken_v8.2.x.js (hunk 1 above)
Comment (facundo-viale-olx) [reply definitions/npm/jsonwebtoken_v8.2.x/flow_v0.56.x-/jsonwebtoken_v8.2.x.js]: done

**Comment by GAntoine** [comment]
@facundo-viale-olx One more thing, you'll need to copy over and teak the specs before I can get this merged in.

**Comment by migueloller** [comment]
Something I'm surprised no one has noticed is that the types for `jwt$Callback` are wrong. Currently they are:

```js
declare type jwt$Callback = (tokenOrError: Error | string) => void;
```

But the function actually takes two arguments, `(error, token)`.

This could be represented with a union of the two signatures. One that takes an error and no token and another that takes null for the error and a string for the token.

**Comment by facundo-viale-olx** [comment]
Done!

**Pull request merged**